[14.0][l10n_br_nfe] mig l10n_br_nfe

rvalyi opened the pull request:
migração padrão do módulo l10n_br_nfe em cima do PR da migração para v13 https://github.com/OCA/l10n-brazil/pull/1673 e incluindo o PR https://github.com/OCA/l10n-brazil/pull/1672 ate rolar o merge dele.

O objetivo é funcionar e passar os testes. Mas alguns warning no Runbot vão dar um pouco mais de trabalho.

mileo committed:
[IMP] Related Document Export

gabrielcardoso21 committed:
[ADD] Export only selected

luismalta committed:
[REM] processador_edoc field definition

rvalyi committed:
avoids NFe errors on test invoice validation

rvalyi committed:
propagate the upstream change: erpbrasil_edoc -> PROCESSADOR_OCA

rvalyi committed:
filter better overrides from invoice validation (fix tests)

rvalyi committed:
simplify import methods according to spec_model_driven

gabrielcardoso21 committed:
[FIX] Related technical_support_id as infRespTec

mileo committed:
[FIX] CFOP Code unmask

gabrielcardoso21 committed:
[FIX] Change comodel name

mileo committed:
[IMP] NF-e Security

mileo committed:
[REM] Fake certificate from l10n_br_nfe

mileo committed:
[REF] generate key on l10n_br_nfe

gabrielcardoso21 committed:
[ADD] Non electronic related fiscal documents

rvalyi committed:
cfop: last merge breaks tests

rvalyi committed:
fix same label warn after new party mixin

rvalyi committed:
more robust nfe filter

renatonlima committed:
Added fields vDesc and vFCPUFDest in NF-e

rvalyi committed:
amount_discount>amount_discount_value (https://github.com/OCA/l10n-brazil/pull/1175)

renatonlima committed:
[FIX] document line vProd import/export

marcelsavegnago committed:
[REM] Test enable certificate on l10n_br_nfe

mileo committed:
[FIX] Flake8 l10n_br_nfe

mileo committed:
fixup! [FIX] Flake8 l10n_br_nfe

mileo committed:
[REM] Processador edoc from main company

mileo committed:
[IMP] Chave, codigo aleatorio, digito verificador e data de entrada e saída

mileo committed:
[FIX] Correção dos testes

mileo committed:
[IMP] Add module version BETA

mileo committed:
[REF] Pagamentos

mileo committed:
[REF] Prepare pagamentos

mileo committed:
[IMP] xmldiff formating

mileo committed:
[IMP] Test with payment

mileo committed:
[FIX] flake8

mileo committed:
[FIX] Make PDF / View

mileo committed:
[FIX] Document number / document confirm and document date

mileo committed:
[FIX] Revisão de erros l10n_br_nfe

mileo committed:
[FIX] Inutilização em produção

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
l10n_br_nfe 12.0.1.1.0

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.1.2.0

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

mileo committed:
[REF] Related docs on l10n_br_nfe

mileo committed:
[REM] fiscal_document_event_ids of l10n_br_nfe

mileo committed:
[REF] Rename invalidate number model on l10n_br_nfe

mileo committed:
[REF] Disable import of document_cancel and document_correction

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

mileo committed:
[REF] Renaming NF-e Wizard refactor

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

mileo committed:
[REF] Renaming NF-e fields

mileo committed:
[FIX] Cancelamento NF-E

mileo committed:
[REF] Event Sequence

mileo committed:
[IMP] Correction letter sequence

mileo committed:
[REF] Invalidate number on l10n_br_nfe

mileo committed:
[IMP] Event file saving / usability / danfe creating

mileo committed:
[FIX] Flake8

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

mileo committed:
[REF] Clean l10n_br_nfe event

mileo committed:
[IMP] Name of document and invalidate

marcelsavegnago committed:
[IMP] l10n_br_nfe: imp authorization_event on invalidate

marcelsavegnago committed:
[IMP] l10n_br_nfe: add correction justification on create event

mileo committed:
[FIX] Tests

mileo committed:
[REM] Document Section of l10n_br_nfe

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

rvalyi committed:
force unique NFe schema stacked field labels

gabrielcardoso21 committed:
[ADD] Base and value ICMSST export

renatonlima committed:
[FIX] vProd with amount_untaxed and discount_value

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.1.3.0

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.1.3.1

mileo committed:
[FIX] l10n_br_nfe return when try to transmit a inutilizate number

mileo committed:
[REF] l10n_br_nfe views

mileo committed:
[REF] Event PROD/HML l10n_br_nfe

mileo committed:
[REF] key -> document_key on l10n_br_nfe

mileo committed:
[REF] number -> document_number on l10n_br_nfe

mileo committed:
[REF] Protect methods of l10n_br_nfe

gabrielcardoso21 committed:
[ADD] Export tag fat

luismalta committed:
[IMP] Abroad recipient

mileo committed:
[FIX] partner cnpj cpf punctuation rm

luismalta committed:
[IMP] res.partner dest condition

mbcosta committed:
[FIX] Field related should inform comodel_name to avoid errors.

luismalta committed:
[ADD] nfe.40.exporta on NFe inherit

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

luismalta committed:
[ADD] Exporta group on Delivery page

mileo committed:
[FIX] Monta processo completo nf-e

mileo committed:
[REF] Document Date to avoid error when install l10n_br_account on nfe

marcelsavegnago committed:
[IMP] l10n_br_nfe: add field nfe40_xEnder

renatonlima committed:
[ADD] NFe line DIFAL fields mapping

renatonlima committed:
[ADD] NFe document DIFAL fields mapping

renatonlima committed:
Remove Difal from icms dict

renatonlima committed:
pep8

OCA-git-bot committed:
l10n_br_nfe 12.0.1.4.0

luismalta committed:
[REM] nfe40_idEstrangeiro

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.0.0

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.1.0

renatonlima committed:
[FIX] nfe40_NFref comodel_name attribute

luismalta committed:
[FIX] Compute for idDest

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.2.0

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.3.0

rvalyi committed:
[FIX] _compute_nfe40_xEnder in res.partner

renatonlima committed:
[FIX] nfe40_cMun related

renatonlima committed:
[REF] l10n_br_nfe: adapt ibge_code

renatonlima committed:
[FIX] readonly=True attribute in nfe40_cMun and nfe40_xMun

renatonlima committed:
pep8

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.4.0

rvalyi committed:
lint: sed -i 's/assertEquals/assertEqual/g'

renatonlima committed:
[FIX] nfe40_uTrib and nfe40_uCom must be related with document line uom fields

mileo committed:
[REF] Standard OCA website l10n_br_nfe

luismalta committed:
[FIX] Add nfe40_choice7 to res.partner

luismalta committed:
[FIX] Depends for _compute_nfe_data

renatonlima committed:
Rename file name

renatonlima committed:
[FIX] match_or_create_m2o in uom object

renatonlima committed:
[FIX] NF-e document line fiscal quantity and remove uom inverse method

renatonlima committed:
[FIX] _compute_nfe40_xEnder method

OCA-git-bot committed:
l10n_br_nfe 12.0.2.4.1

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.4.2

luismalta committed:
[IMP] Definition of "idEstrangeiro" in nfe40_choice7

luismalta committed:
[FIX] Foreign partner validation

luismalta committed:
[FIX] Return value for nfe40_idEstrangeiro

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.2.4.3

ygcarvalh committed:
[REM] Unnecessary code

ygcarvalh committed:
[REM] Unnecessary code

ygcarvalh committed:
[IMP] 'Simples Nacional' xml tags

ygcarvalh committed:
[IMP] 'Simples Nacional' xml tags

ygcarvalh committed:
[IMP] NF-e Serialize test function

ygcarvalh committed:
[ADD] 'Lucro Presumido' test file functions

ygcarvalh committed:
[ADD] 'Simples Nacional' test file functions

ygcarvalh committed:
[ADD] Test files

ygcarvalh committed:
[ADD] XML Files

ygcarvalh committed:
[IMP] Demo for new tests

gabrielcardoso21 committed:
[FIX] Size of description

mileo committed:
[FIX] Validate payment information

ygcarvalh committed:
[FIX] Flake8

OCA-git-bot committed:
l10n_br_nfe 12.0.2.5.0

gabrielcardoso21 committed:
[FIX] Name with \n character

gabrielcardoso21 committed:
[FIX] Rejection 881

gabrielcardoso21 committed:
[FIX] Export NFref

renatonlima committed:
[FIX] Not copy NF-e env and version always get from default

renatonlima committed:
[REF] remove duplicated code

renatonlima committed:
[REF] partner fix inverse method, remove unsed field

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

rvalyi committed:
_compute_* should assign all records

rvalyi committed:
[IMP] l10n_br_nfe: black, isort, prettier

rvalyi committed:
v14 compat: no more field._attrs in v14

rvalyi committed:
temporary flake8 disable C901 too complex meth

renatonlima committed:
[REF] l10n_br_nfe: rename amount_financial to amount_financial_total

OCA-git-bot committed:
l10n_br_nfe 12.0.2.5.1

rvalyi committed:
adjust pre-commit after rebase

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.3.0.0

renatonlima committed:
[FIX] Quando for atualizado o campo cnpj_cpf deve ser formatado.

renatonlima committed:
[FIX] Implements compute and inverse for nfe40_IE

renatonlima committed:
[FIX] NFe import test

OCA-git-bot committed:
l10n_br_nfe 12.0.3.1.0

gabrielcardoso21 committed:
[FIX] Export of tag vTroco

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

marcelsavegnago committed:
[FIX] l10n_br_nfe: fix ipi test for SN company

marcelsavegnago committed:
[FIX] l10n_br_nfe: fix test_nfe_serialize.py

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

rvalyi committed:
fixes #1559 - disable buggy detpag validation #1457

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

renatonlima committed:
[FIX] fiscal document related Key

renatonlima committed:
[ADD] refCTe field in fiscal document related

renatonlima committed:
[ADD] nfe_transmission, nfe tpEmis company default field

renatonlima committed:
[ADD] Fiscal document print layout for NF-e and NFC-e

renatonlima committed:
[REF] adapt nfe40_Id

renatonlima committed:
l10n_br_nfe: pre-commit

renatonlima committed:
[FIX] removed unused inherits

renatonlima committed:
pre-commit

rvalyi committed:
[FIX] adapted tests

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.3.2.0

renatonlima committed:
pre-commit

renatonlima committed:
Rename XML's files

renatonlima committed:
[FIX] NFe compute data nfe40_id, nfe40_tpImp

renatonlima committed:
[FIX] l10n_br_nfe hook

renatonlima committed:
[FIX] fiscal document NFe demo data

renatonlima committed:
[FIX] nfe tests

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

antoniospneto committed:
fix estimate_tax variable name

antoniospneto committed:
fix tests

antoniospneto committed:
fix street_number in l10n_br_nfe

antoniospneto committed:
fix andress fields

antoniospneto committed:
fix demo data

renatonlima committed:
Set null value for nfe40_Id

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

rvalyi committed:
[REF] remove useless @api.multi

rvalyi committed:
[MIG] l10n_br_nfe: Migration to 13.0

rvalyi committed:
[FIX] add missing erpbrasil.edoc.pdf dep

rvalyi committed:
[FIX] avoids CacheMiss error

rvalyi committed:
[MIG] l10n_br_nfe: Migration to 14.0

crsilveira commented:
Tentando ajudar, instalei o l10n_br_nfe, e estou tendo problema quando adiciono um item na nfe,ajuda reportar isso aqui ? Se sim, coloco todas as informações.

rvalyi commented:
legal @crsilveira tou focado na migraçaod do l10n_br_account para destravar os outros modulos, mas devo conseguir retormar isso essas dias para deixar funcional como na v13. Deve ser alguma besteirinha porque umas semanas atras antes dos ultimos rebases ja conseguimos emitir Nfe's na 14 basicamente com essas mudanças...

crsilveira commented:
blz @rvalyi ,então instalei tudo sem problema usando os dados demo, o problema acontece quando cria uma nfe, se colocar algum produto, não  SALVA, dá o erro abaixo:

odoo/fields.py", line 1341, in convert_to_cache
    return float_round(value, precision_digits=digits[1]) if digits else value
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "../odoo14/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "../odoo14/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: 'int' object is not subscriptable

crsilveira commented:
Não sei se ajuda, está dando este mesmo erro quando tento Confirmar, :
l10n-brazil/l10n_br_nf < Location    >│
  432             if │e/models/document_line.py", line 441, in _export_fields                                                                        │
  433                │    self.nfe40_qCom = self.quantity

in convert_to_cache                                                                                                            │
  1335         # appl│    return float_round(value, precision_digits=digits[1]) if digits else value                                                 │
  1336         value │TypeError: 'int' object is not subscriptable

antoniospneto committed:
fix fiscal_operation_type False validation

oca-travis committed:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
l10n_br_nfe 12.0.3.2.1

rvalyi committed:
[FIX] digits=(a, b) for Odoo v14 compat

fix in generateds-odoo: https://github.com/akretion/generateds-odoo/commit/7918e14cae05c1b0a32da57223a82070f992976f

rvalyi committed:
[FIX] pCredSN is filtered out from codegen

ICMS* and ICMSSN* fields are skipped from the l10n_br_nfe_spec generation
(to avoid ICMS*/ICNSSN* fields collision). They don't need to exist
as Odoo fields and are dealt with by the import/export mapping methods.

rvalyi commented:
Ola, fiz um rebase do PR com alguns fixes.
@crsilveira o erro que vc reportou foi corrigido por esse commit https://github.com/OCA/l10n-brazil/pull/1674/commits/7013ba362897586c521e83822e3908183553f445 no l10n_br_nfe_spec, pois na v12 voce podia ter um fields.Float com digits=4 e na v14 tem que ser digits=(16, 4). O Odoo não usa o primeiro digito na vdd, mas tem que ser nesse formato. O gerador com generateDS foi alterado tb https://github.com/akretion/generateds-odoo/commit/7918e14cae05c1b0a32da57223a82070f992976f

Por algum motivo que eu nao entendo eu tive que fazer essa alteraçao temporaria: https://github.com/OCA/l10n-brazil/pull/1674/commits/fb3af4cf1a8771025634f3b750297df005f00caa Sem alguem entende como evitar esse store=True eu agradeço!

Nessa altura os testes so não estão passando porque foi desativada a criaçao automatica dos certificados fake (que alias nem permitem transmitir para a fazenda em homologaçao) na migraçao do l10n_br_fiscal porque usava coisas deprecadas que gerava warning no Runbot da 14: https://github.com/OCA/l10n-brazil/commit/81eb68aaa28a7f6e94ea731d9757c841cc8a0252

Fora esse problema nos tests, o modulo deve funcionar em prod.

renatonlima committed:
[ADD] ir.config.parameter for nfe VerProc tag

renatonlima committed:
[FIX] inconsistent 'compute_sudo' in res.partner nfe40 compute fields

renatonlima committed:
[FIX] ICMSUFDest NFE tag

crsilveira commented:
l10n_br_nfe/models/document_line.py", line 388, in _export_fields
    icms_tag = self.nfe40_choice11.replace("nfe40_", "")  # FIXME

tentando ver os problemas ..., quando cria a fatura está criando 2 linhas na l10n_br_fiscal_document_line, isso faz com que dê erro na linha acima, pois,  a linha de crédito não tem o produto nela ... e ao validar a Fatura e Criar o XML , está criando uma lnha no XML desta segunda linha...  onde eu poderia evitar q esta segunda linha entre no XML ?

rvalyi commented:
fiz um rebase, continua com o problema dos certificados nos testes mencionado aqui https://github.com/OCA/l10n-brazil/pull/1674#issuecomment-963238162 e que tera que resolver no pacote erpbrasil.assinatura.

renatonlima committed:
[FIX] Remove default from compute field

renatonlima committed:
[REF] rename line_ids to fiscal_line_ids

renatonlima committed:
Avoid warning message: selection attribute will be ignored as the field is related

marcelsavegnago commented:
@rvalyi pode fazer um rebase por favor ? vlww

marcelsavegnago commented:
> l10n_br_nfe/models/document_line.py", line 388, in _export_fields icms_tag = self.nfe40_choice11.replace("nfe40_", "") # FIXME
> 
> tentando ver os problemas ..., quando cria a fatura está criando 2 linhas na l10n_br_fiscal_document_line, isso faz com que dê erro na linha acima, pois, a linha de crédito não tem o produto nela ... e ao validar a Fatura e Criar o XML , está criando uma lnha no XML desta segunda linha... onde eu poderia evitar q esta segunda linha entre no XML ?

eu peguei para testar agora e cai nesta mesma questão.

marcelsavegnago committed:
[FIX] l10n_br_nfe: fix size nfe40_verProc value

marcelsavegnago commented:
@rvalyi se puder fazer aquele rebase eu agradeço :) ❤️

rvalyi committed:
[MIG] Odoo Brasil v12.0 -> Odoo Brasil OCA v14

rvalyi committed:
[MIG] line_ids -> fiscal_line_ids

rvalyi committed:
[MIG] v14 HACK TODO FIXME!

rvalyi committed:
pre-commit

rvalyi commented:
@marcelsavegnago @renatonlima @netosjb @felipemotter @vanderleiromera @hirokibastos @mbcosta 

fiz o rebase e ta verdinho. Porem:
- eu tive que refazer esse HACK feio https://github.com/OCA/l10n-brazil/pull/1674/commits/e6fb7396fa6b2fefa12d57dbd24ffb180fb270c5
- vai ter que refazer os cherry-picks desde Novembro 2021. Deve ter uns 20 commits talvez, não é nada demais hoje.
- tem esses WARNINGS https://gist.github.com/rvalyi/b420639ff6b2652070836be166750cae coisas de campo seleção. Eu sugiro de passar por cima no momento em vez de deixar essa PR aberta por anos.

Com tudo minha sugestão é fazer o merge: nisso a gente mata 80% do misterio dessa PR para quem se interessa em NFe na 14 e fica fácil alguém completar pelo trem de forward port de cherry-picks do que falta da v12 no l10n_br_nfe. Os warnings a gente pode ver o que a gente consegue solucionar, mas talvez uma parte morre tb quando a gente troca o generateDS pelo xsdata daqui pouco na frente. A meu ver a coisa mais chata para resolver sera tirar esse HACK  https://gist.github.com/rvalyi/b420639ff6b2652070836be166750cae mas o impacto vai ser apenas de criar alguns campos em algumas tabelas que deveria ser apenas calculados. Eu quebrei a cabeça varias horas nisso ainda sem conseguir resolver mas eu tenho umas ideias de melhoria que eu irei desenvolver em outro PR.

Bom, OK para fazer o merge dessa PR para facilitar de melhorar a NFe na 14?

rvalyi commented:
/ocabot merge nobump
